.github/workflows/deploy.yaml

```diff
@@ -4,15 +4,46 @@ on:
       - master
       - develop
 
-env:
-  JAVA_VERSION: "11.x"
-  FLUTTER_VERSION: "2.0.1"
-  PROPERTIES_PATH: "android/key.properties"
-
 jobs:
+  bump_version:
+    name: "Bump pubspec version"
+    runs-on: ubuntu-latest
+    env:
+      APP_VERSION_PATH: "app_feup/app_version.txt"
+      PUBSPEC_PATH: "app_feup/pubspec.yaml"
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.NIAEFEUPBOT_PAT }}
+          fetch-depth: 2
+
+      - name: Ensure app_version was not changed manually or was created just now
+        run: |
+          (! git cat-file -e ${{ github.ref_name }}^:${{ env.APP_VERSION_PATH }} && git cat-file -e HEAD:${{env.APP_VERSION_PATH}}) || git diff --quiet ${{ github.ref_name }}^:${{ env.APP_VERSION_PATH }} HEAD:${{ env.APP_VERSION_PATH }} || exit 1
```

Comment:
Can  we add a small log to make it more visible why this failed?

Reply:
For sure, should I simply dump it to the console?